src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -584,26 +626,22 @@ public function create_item( $request ) {
 			);
 		}
 
-		// Do not allow comments to be created with a non-default type.
-		if ( ! empty( $request['type'] ) && 'comment' !== $request['type'] ) {
-			return new WP_Error(
-				'rest_invalid_comment_type',
-				__( 'Cannot create a comment with that type.' ),
-				array( 'status' => 400 )
-			);
-		}
-
 		$prepared_comment = $this->prepare_item_for_database( $request );
 		if ( is_wp_error( $prepared_comment ) ) {
 			return $prepared_comment;
 		}
 
-		$prepared_comment['comment_type'] = 'comment';
+		$prepared_comment['comment_type'] = $request['type'];
```

Comment:
```suggestion
		$prepared_comment['comment_type'] = empty( $request['type'] ) ? 'comment' : $request['type'];
```

When `type` isn't specified fallback to `comment` like before. The validation above should handle unsupported types.

Suggestion based on feedback: https://github.com/WordPress/wordpress-develop/pull/10280/files#r2432329836